CONCURRENCY.md

```diff
@@ -149,3 +178,32 @@ the following mechanisms to prevent the unintended sharing of state via global o
    * enums are always frozen
 
  Combined, these mechanisms allow natural race-freeze programming with code reuse across platforms in MPP projects.
+
+<a name="atomic_references"></a>
+### Atomic primitives and references
+
+ Kotlin/Native standard library provides primitives for safe working with concurrently mutable data, namely
+`AtomicInt`, `AtomicLong`, `AtomicNativePtr`, `AtomicReference` and `FreezableAtomicReference` in the package
+`kotlin.native.concurrent`.
+Atomic primitives allows concurrency-safe update operations, such as increment, decrement and compare-and-swap,
+along with value setters and getters. Atomic primitives are considered always frozen by the runtime, and their fields cannot
+be updated with the regular field mutation, but the value can be changed using dedicated operations, so it is possible
```

Comment:
> cannot be updated with the regular field mutation

`atomicInt.value` mutation with getter and setter is not distinguishable from "regular field mutation" at call site.
Moreover, `atomicInt.value += 1` is still possible, and it is not too different from "regular field mutation" by its behaviour.